astropy/io/misc/hdf5.py

```diff
@@ -213,6 +213,26 @@ def _encode_mixins(tbl):
 
     return encode_tbl
 
+def check_column_encoding(table):
+    """
+    Table with native py3 strings can't be stored in a hdf5 format so if
+    table has py3 strings then a copy of table is made and stored in temp.
+    the copy stored in temp has columns in bytes unicode.
+    """
+    from astropy.table import Table
+    col = []
+    temp = Table()
```

Comment:
No need to create an empty table here as you are creating a temporary copy below if needed.

Reply:
fine I'll remove that line